src/librustc_trans/save/data.rs

```diff
@@ -0,0 +1,351 @@
+// Copyright 2012-2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use middle::ty;
+use middle::def_id::DefId;
+
+use std::hash::Hasher;
+use syntax::ast::{self, NodeId};
+use syntax::codemap::Span;
+
```

Comment:
Can you add a comment to the file stating what is here? And perhaps move normalize_node_id lower in the file, since it is a utility function